llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUInstPrinter.cpp

```diff
@@ -539,8 +539,17 @@ void AMDGPUInstPrinter::printImmediate16(uint32_t Imm,
   }
 
   uint16_t HImm = static_cast<uint16_t>(Imm);
-  if (printImmediateFloat16(HImm, STI, O))
-    return;
+  switch (OpType) {
+  case AMDGPU::OPERAND_REG_IMM_FP16:
```

Comment:
Does it need to handle bf16 here? It is somewhat weird to have `printImmediate16` and only handle i16 and f16. Or should it be renamed into `printImmediateF16` and have no `OpType` and this switch, given its only use?